### PR TITLE
feat: Implement even more additional overhead button/switch functionality (MASK MAN ON, EMER ELEC PWR MAN ON, Cargo smoke DISCH, etc.)

### PR DESCRIPTION
**Pull request by ErickSharp**
## Summary of Changes

* Implemented OXYGEN MASK MAN ON overhead functionality
    * OXYGEN MASK MAN ON cover animation and button press
    * Oxygen passenger annunciator
    * Oxygen timer reset functionality
* Implemented EMER ELEC PWR button functionality
    * Made EMER GEN TEST button pressable
    * Made MAN ON button pressable
    * Removed RAT & EMER GEN annunciator cursor 
* Implemented RCDR button functionality
    * GND CTL button pressable
    * CVR ERASE button pressable
    * CVR TEST button pressable
* Implemented misc. overhead maintenance panel button functionality 
    * SVGE INT OVRD button pressable
    * AVIONICS COMP LT button pressable
    * APU AUTO EXITING TEST button pressable
    * APU AUTO EXITING RESET button pressable 
* Implemented Data Loading Selector button functionality
    * NEXT button pressable
    * PREV button pressable
    * SEL CTL button pressable
    * ON/OFF switch animation
* Implemented weather radar switch functionality
    * GCS switch animation
    * MULTISCANS switch animation
* Implemented ELT button functionality
    * ELT ON button pressable
    * ELT TEST/RESET button pressable
* Implemented FWD and AFT Cargo Smoke DISCH button functionality
* Removed (INOP.) from TOILET and PA annunciators
* Changed cursors of circular buttons to a "hand" (pointer) cursors and the flap lever cursor to a "grab" cursor. 
* Added appropriate tooltips to each button 

## References

*ELT
https://www.youtube.com/watch?v=Bb58tws6Ums&t=5s
*OXYGEN MASK MAN ON 
https://www.youtube.com/watch?v=nE-n2tNy7Io&t=607s
Asked Tarek & Dan for help on some buttons

Other buttons were a matter of following closely the behavior of related buttons or trying to make a very close approximation as with the case of covered push buttons like the OXYGEN MASK MAN ON and EMER ELEC PWR MAN ON, as these buttons are actually modeled incorrectly.  

## Additional context

Discord username (if different from GitHub): Erick [Z-6]#6484

## Testing instructions

Check that the buttons outlined in the SUMMARY OF CHANGES section are clickable, their tooltips are appropriate, and that the cursor for the flap lever is now a grab, and that the cursor for the small buttons are pointers.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: .github/CHANGELOG.md
Comment (beheh) [inline .github/CHANGELOG.md:76]: You have merge conflict markers in this file. Please rebase and fix those.

**Review by beheh — CHANGES_REQUESTED** [review]

**Review by beheh — APPROVED** [review]

**Comment by stescouse** [comment]
Category: QA Tester
Name: stescouse#7271
Date of testing: 13/2/21
Version of the sim: 1.12.13.0
PR Tested: #3376
Tier of testing: 1
Changes to observe: 

Testing technique: TryOut
Steps you've made to test the Pull Request:

1. Spawned C&D at SBGL
2. Pressed  OXYGEN MASK MAN ON overhead function and observed OXYGEN MASK MAN ON cover animation and button press.  Observed Oxygen passenger annunciator.  Pressed Oxygen timer reset and observed functionality.

OXEGEN 1

OXEGEN 2


3. Pressed EMER ELEC PWR button functionality, pressed EMER GEN TEST button observed functionality. Pressed MAN ON button observed functionality. Observed change in RAT & EMER GEN annunciator cursor

e n1


4. Pressed RCDR button, GND CTL button, CVR ERASE button, and CVR TEST button and observed functionality

record

5. Pressed SVGE INT OVRD button, AVIONICS COMP LT button, APU AUTO EXITING TEST button, and APU AUTO EXITING RESET button and observed functionality.

avonics 1

apu 1

6.  Observed  Data Loading Selector button functionality, pressed NEXT button, PREV button, SEL CTL button.  Observed ON/OFF switch animation.

data 1


6. Observed weather radar switch functionality, pressed GCS switch and MULTISCANS switch.

weather 1


7. Observed ELT button functionality.  Pressed ELT ON button and ELT TEST/RESET button. 

8. Observed FWD and AFT Cargo Smoke DISCH button functionality.

smoke


9. Observed 1 and 2 ENG N1 MODE button functionality

10. Observed (INOP.) removed from TOILET and PA annunciators

11.  Observed changed cursors of circular buttons to a "hand" (pointer) cursors and the flap lever cursor to a "grab" cursor.
and appropriate tooltips to each button


Flight notes (If required: none

Issues (If Any): None


Overall Rating: Good

Conclusions of the testing: performed as outlined in PR.

**Comment by marcman86** [comment]
Name: marcman86#4907
Date of testing: 13.2.2021
Version of the sim: 1.12.13.0
PR Tested:  #3376
Tier of testing: 2
Changes to observe: Look at new implemented buttons
Testing technique: Tryout

Steps you've made to test the Pull Request:

1.) Spawned at day C&D EDDP
2.) Turn on bat. and external pwr.

3.) Functionality check (Animation, Sound, Decal) 

OXYGEN MASK MAN ON


OXYGEN TMR RESET


EMER ELEC PWR


RCDR


MISC



Data Loading Selector


Weather Radar Switches


ELT

Cargo Smoke DISCH


1 and 2 ENG N1 MODE


(INOP.) from TOILET

flap lever "grab" cursor


Issues (If Any): None

Overall Rating: Good

Conclusions of the testing: Works!

Hope this helps :-)

**Comment by Aarondodd88** [comment]
Awesome job.  GND CTL turned on during pre-flight, the aim here is that the voice recorder is operating when checklist is read, GND CTL button switches to auto once an engine is started.   Currently you have to press GND CTL again for it to turn off, even after both engines have started

**Comment by ErickSharp** [comment]
> Awesome job. GND CTL turned on during pre-flight, the aim here is that the voice recorder is operating when checklist is read, GND CTL button switches to auto once an engine is started. Currently you have to press GND CTL again for it to turn off, even after both engines have started

You are saying that after the engine start, the ground control button should go to auto?

**Comment by Aarondodd88** [comment]
Yup. GND CTL button to auto when engine starts http://a320dp.com/A320_DP/communications/sys-2.5.0.html

**Comment by ErickSharp** [comment]
> Yup. GND CTL button to auto when engine starts http://a320dp.com/A320_DP/communications/sys-2.5.0.html

Ok, so is this button "momentary?" When you press it, does it change and then just pop back out or does it stay in? After the engine start, does this button automatically switch to auto and then the "on" light comes off?

**Comment by GlitchedYT** [comment]
> > Yup. GND CTL button to auto when engine starts http://a320dp.com/A320_DP/communications/sys-2.5.0.html
> 
> Ok, so is this button "momentary?" When you press it, does it change and then just pop back out or does it stay in? After the engine start, does this button automatically switch to auto and then the "on" light comes off?

So the button stays in when you press it, getting info from danbw9 to see when it automatically turns off.

**Comment by ErickSharp** [comment]
> > > Yup. GND CTL button to auto when engine starts http://a320dp.com/A320_DP/communications/sys-2.5.0.html
> > 
> > 
> > Ok, so is this button "momentary?" When you press it, does it change and then just pop back out or does it stay in? After the engine start, does this button automatically switch to auto and then the "on" light comes off?
> 
> So the button stays in when you press it, getting info from danbw9 to see when it automatically turns off.

Any word on this?

**Comment by GlitchedYT** [comment]
So I have found out by danbw9 that once engines start the grn ctrl button depresses. The only thing that is odd is that I remember with the fslabs it didn’t turn off until after pushback when the ground crew disconnected but that could be wrong.

**Comment by ErickSharp** [comment]
> So I have found out by danbw9 that once engines start the grn ctrl button depresses. The only thing that is odd is that I remember with the fslabs it didn’t turn off until after pushback when the ground crew disconnected but that could be wrong.

He told me that the button is in fact momentary. If I do choose to add this functionality before this PR gets pulled, I will have to find out how to toggle the simvar when one engine turns on.

**Comment by Aarondodd88** [comment]
Yup once engine is on it doesn't need to energize itself. I have faith in you :)

**Comment by beheh** [comment]
Would be good to have one more QA test after the latest changes before merging this.

**Comment by marcman86** [comment]
Tested again. PR is like before. GND CTL turnes not to AUTO when engines are AVAIL.

**Comment by wpine215** [comment]
Before this PR can be merged, ENG N1 mode button functionality should be removed. A320neo's with CFM LEAP engines do not have these buttons - I have forwarded a request to Asobo to address this model issue in a future update, but in the meantime, I think keeping these buttons INOP is the best decision.

**Comment by ErickSharp** [comment]
> Before this PR can be merged, ENG N1 mode button functionality should be removed. A320neo's with CFM LEAP engines do not have these buttons - I have forwarded a request to Asobo to address this model issue in a future update, but in the meantime, I think keeping these buttons INOP is the best decision.

Yep I remember seeing that. Aye Aye, Captain.

**Comment by ErickSharp** [comment]
> Before this PR can be merged, ENG N1 mode button functionality should be removed. A320neo's with CFM LEAP engines do not have these buttons - I have forwarded a request to Asobo to address this model issue in a future update, but in the meantime, I think keeping these buttons INOP is the best decision.

Done.

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (beheh) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:2824]: This is a typo (#3531), let's update the comment:

```suggestion
                    <!-- APU AUTO EXTING TEST-->
```

**Comment by marcman86** [comment]
ENG N1 are iluminate like in ANN LT test, even ANN LT lever is not in TEST position.

**Comment by ErickSharp** [comment]
> ENG N1 are iluminate like in ANN LT test, even ANN LT lever is not in TEST position.
> 

Fixed. Sorry about that!

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 27.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3376
Tier of testing: 1
Changes to observe: More buttons
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Turned on bat. and ext. power
3.) Tested buttons
4.) Did ANN LT Test

Issues: -

Overall Rating: Good

Conclusions of the testing: Works (sounds, animations and lights)

Hope this helps :-)

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 28/02/2021
Version of the sim: 1.13.16.0
PR Tested: #3376 
Tier of testing: 1
Changes to observe: Additional buttons
Testing technique: 
Steps you've made to test the Pull Request:

1. Loaded C&D
2. Setup the plane
3. Tested all buttons lined out in the PR

Media:

Issues (If Any): All buttons lined out in this PR that are under a safety guard should be permament buttons, not push and release buttons. Also they dont have ON/AUTO etc. indications. Did this get addressed with this PR? 

Overall Rating: LGTM

Conclusions of the testing: If above issues are outside this PR goal, everything works as described

**Comment by ErickSharp** [comment]
> Category: QA Tester
> Name: St54Kevin#3583
> Date of testing: 28/02/2021
> Version of the sim: 1.13.16.0
> PR Tested: #3376
> Tier of testing: 1
> Changes to observe: Additional buttons
> Testing technique:
> Steps you've made to test the Pull Request:
> 
> 1. Loaded C&D
> 2. Setup the plane
> 3. Tested all buttons lined out in the PR
> 
> Media:
> 
> Issues (If Any): All buttons lined out in this PR that are under a safety guard should be permament buttons, not push and release buttons. Also they dont have ON/AUTO etc. indications. Did this get addressed with this PR?
> 
> Overall Rating: LGTM
> 
> Conclusions of the testing: If above issues are outside this PR goal, everything works as described

They actually shouldn't the buttons that Asobo modeled are of the incorrect type. They should be the exact same type of those used for the "CALL" buttons. This is why there is no emissive attached for them, and why they are momentary, meaning they don't pop back out.

**Comment by beheh** [comment]
@Benjozork You've done more XML than I have. Any concerns or are you good with the changes from this PR?

@ErickSharp Could you please fix the conflict with ` docs/a320-simvars.md `? Then we can merge this once we're ready.

**Comment by ErickSharp** [comment]
> @Benjozork You've done more XML than I have. Any concerns or are you good with the changes from this PR?
> 
> @ErickSharp Could you please fix the conflict with `docs/a320-simvars.md`? Then we can merge this once we're ready.

Done deal! :D

File: docs/a320-simvars.md
Comment (lukecologne) [inline docs/a320-simvars.md:530]: This probably shouldn't be in here, since the simvar isn't actually being set/used.

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (lukecologne) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:2159]: Why do some of the attributes you added have a space around the =? Please follow the rest of the file and normal XML formatting

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (lukecologne) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:2697]: Indentation problem

**Review by lukecologne — CHANGES_REQUESTED** [review]
Since most of the buttons you make clickable here don't actually have any functionality at moment, you should add Inop. to the tooltips so that users don't get the idea that the button does anything.

**Comment by ErickSharp** [comment]
Yes, good point. I have addressed these issues @lukecologne

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml
Comment (lukecologne) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml]: There are some more occurences of this

Comment (ErickSharp) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml]: Fixed the other 5 or so occurrences.

**Review by lukecologne — APPROVED** [review]

**Comment by beheh** [comment]
Waiting for one more QA test after the latest changes.

**Comment by DavidEGTE** [comment]
Category: QA Tester
Name: David#2177
Date of testing: 04/03/2021
Version of the sim: 1.13.17.0
PR Tested: #3376
Tier of testing: 2
Changes to observe:
N/A

Testing technique: Try-out

Steps you've made to test the Pull Request:
1. Spawned C&D at LFPG
2. GRD PWR + BAT
3. Started test of all functionality's described on this PR 

Functionality notes: 
Removed (INOP.) from TOILET and PA annunciators <- True
Changed cursors of circular buttons to a "hand" (pointer) cursors and the flap lever cursor to a "grab" cursor. <- True
Added appropriate tooltips to each button <- True

Media:
Microsoft Flight Simulator Screenshot 2021 03 04 - 11 09 41 55
Microsoft Flight Simulator Screenshot 2021 03 04 - 11 10 18 33
Microsoft Flight Simulator Screenshot 2021 03 04 - 11 10 52 51

Issues (If Any): N/A

Steps to achieve the issue
N/A

Overall Rating: Good
Conclusions of the testing: Everything mentioned on PR works

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 04/03/2021
Version of the sim: 1.13.17.0
PR Tested: #3376 
Tier of testing: 1
Changes to observe: New functional overhead buttons

Testing technique: Tryout

Steps you've made to test the Pull Request:

1. Spawned on runway at VABB
2. Tested all the pressable switches mentioned on the list above
3. Turned on ANN LT to test, checked new annunciator lights
4. Checked the hand pointer on single flip-switches, flap handles etc.

Issues (If Any): None

Overall Rating: Good

Conclusions of the testing: New buttons were made pressable, and their functionality and lights were checked and verified. Good to go.

**Pull request merged**
